nest_local_flask/benchmark_plots.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.axes as axes
+import matplotlib.pylab as pylab
+
+import nestpy 
+from interaction_keys import GetInteractionObject
+
+v_nestpy = str(nestpy.__version__)
+v_nest = '2.0.1'
+version_textbox = " NEST v{0} \n nestpy v{1}".format(v_nest, v_nestpy)
+
+# Figure parameters common throughout all plots
+bbox = dict(boxstyle="round", fc="1.00", edgecolor='none')
+params = {'xtick.labelsize':'x-large',
+         'ytick.labelsize':'x-large',
+         }
+pylab.rcParams.update(params)
+
+fields = [1, 2, 5, 10, 20, 50, 100, 200, 500, 1000, 2000, 5000, 10000, 20000]
+energies = np.logspace(-1, 4, 2000)
+energies = np.broadcast_to(energies, (len(fields), len(energies)))
+
+#Detector identification
+detector = nestpy.DetectorExample_XENON10()
+# Performing NEST calculations according to the given detector example       
+nc = nestpy.NESTcalc(detector)
+#Once you have interaction, you can get yields
+#Begin with np.vectorize so that we can get yields over a range of interaction types/energies/drift fields.
+
+@np.vectorize
+def GetYieldsVectorized(interaction, yield_type, **kwargs):
+    # This function does nc.GetYields for the various interactions and arguments we pass into it
+    # TODO: Look at docstrings
+    interaction_object = GetInteractionObject(interaction)
+    if 'energy' in kwargs.keys():
+        if interaction_object == GetInteractionObject('nr') and kwargs['energy'] > 3e2:
+            return np.nan
+        if interaction_object == GetInteractionObject('gammaray') and kwargs['energy'] > 3e3:
+            return np.nan
+        if interaction_object == GetInteractionObject('beta') and kwargs['energy'] > 3e3:
+            return np.nan     
+    yield_object = nc.GetYields(interaction = interaction_object, **kwargs)
+    # returns the yields for the type of yield we are considering 
+    return getattr(yield_object, yield_type)
+
+def PhotonYield(**kwargs):
+    return GetYieldsVectorized(yield_type = 'PhotonYield', **kwargs)
+ 
+def ElectronYield(**kwargs):
+    return GetYieldsVectorized(yield_type = 'ElectronYield', **kwargs)
+
+def Yield(**kwargs):
+    return {'photon' : PhotonYield(**kwargs),
+            'electron' : ElectronYield(**kwargs),
+           # What is missing?  Aren't there other parts of YieldObject?
+           }
+
+#Calculate yields for various benchmark calculations
```

Comment:
`kwargs = {'energy': energies.T, 'drift_field': fields}`

Then replace below with:

`ElectronYield(interaction='nr', **kwargs)/energies`

Could energies be done somewhere else for less repetition?  Lots of repetition here.